src/rdkafka_broker.c

```diff
@@ -369,14 +369,28 @@ static void rd_kafka_broker_set_error (rd_kafka_broker_t *rkb, int level,
                                        const char *fmt, va_list ap) {
         char errstr[512];
         char extra[128];
-        size_t of, ofe;
+        size_t of = 0, ofe;
         rd_bool_t identical, suppress;
         int state_duration_ms = (int)((rd_clock() - rkb->rkb_ts_state)/1000);
 
 
-        of = (size_t)rd_vsnprintf(errstr, sizeof(errstr), fmt, ap);
-        if (of > sizeof(errstr))
-                of = sizeof(errstr);
+        /* If this is a logical broker we include its current nodename/address
+         * in the log message. */
+        rd_kafka_broker_lock(rkb);
+        if (rkb->rkb_source == RD_KAFKA_LOGICAL && *rkb->rkb_nodename) {
```

Comment:
side note: the idea of re-using the `rd_kafka_broker_t` type for something that isn't linked to a physical broker seems jarring to me. i say this without specific thoughts about the requirements, but i'm guessing the potential here is significant, given the scope of what a controller does. specialized additional type(s) would provide better clarity and encapsulation.